### ISSUE TITLE
uWSGI and Python's sys.stdout.detach()

### ISSUE THREAD

**Issue by PAStheLoD** [unbit/uwsgi#577]
Hello!

I've run into a very strange (but maybe not completely unexpected) problem.

This
```
import codecs
import logging
import sys

import flask

app = flask.Flask(__name__)

logging.getLogger().setLevel(logging.DEBUG)
logging.debug('ok')

s = sys.stdout.detach()
sys.stdout = s

logging.debug('not ok')
```

fails on uWSGI without any explanation

```
root@9143656d4a36:/opt/# uwsgi -w i:app --http 0.0.0.0:8080
*** Starting uWSGI 2.0.3 (64bit) on [Sun Mar 23 23:02:49 2014] ***
compiled with version: 4.8.2 on 23 March 2014 21:58:57
os: Linux-3.13.0-18-generic #38-Ubuntu SMP Mon Mar 17 21:40:06 UTC 2014
nodename: 9143656d4a36
machine: x86_64
clock source: unix
detected number of CPU cores: 2
current working directory: /opt/
detected binary path: /usr/local/bin/uwsgi
!!! no internal routing support, rebuild with pcre support !!!
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
*** WARNING: you are running uWSGI without its master process manager ***
your processes number limit is 15849
your memory page size is 4096 bytes
detected max file descriptor number: 1024
lock engine: pthread robust mutexes
thunder lock: disabled (you can enable it with --thunder-lock)
uWSGI http bound on 0.0.0.0:8080 fd 4
spawned uWSGI http 1 (pid: 2632)
uwsgi socket 0 bound to TCP address 127.0.0.1:43414 (port auto-assigned) fd 3
Python version: 3.3.5 (default, Mar 11 2014, 02:52:08)  [GCC 4.8.2]
*** Python threads support is disabled. You can enable it with --enable-threads ***
Python main interpreter initialized at 0x9eb770
your server socket listen backlog is limited to 100 connections
your mercy for graceful operations on workers is 60 seconds
mapped 72752 bytes (71 KB) for 1 cores
*** Operational MODE: single process ***
DEBUG:root:ok
unable to load app 0 (mountpoint='') (callable not found or import error)
*** no app loaded. going in full dynamic mode ***
*** uWSGI is running in multiple interpreter mode ***
spawned uWSGI worker 1 (and the only) (pid: 2631, cores: 1)
```

but works with vanila python.

```
root@9143656d4a36:/opt/# python3.3 i.py 
DEBUG:root:ok
DEBUG:root:not ok
```

**Comment by xrmx** [comment]
I think we are good to close this.
